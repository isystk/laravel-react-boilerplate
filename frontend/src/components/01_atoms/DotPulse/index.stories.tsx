import { Meta, Story } from '@storybook/react'
import React from 'react'
import DotPulse from './index'

export default {
  title: '01_atoms/DotPulse',
  component: DotPulse,
} as Meta

const Template: Story = () => {
  return <DotPulse />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
