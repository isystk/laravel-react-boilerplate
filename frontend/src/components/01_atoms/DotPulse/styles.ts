import { css, keyframes } from '@emotion/css'

const dotPulseBefore = keyframes`
0% {
  box-shadow: 9984px 0 0 -5px #9880ff;
}
30% {
  box-shadow: 9984px 0 0 2px #9880ff;
}
60%, 100% {
  box-shadow: 9984px 0 0 -5px #9880ff;
}
`

const dotPulseCenter = keyframes`
0% {
  box-shadow: 9999px 0 0 -5px #9880ff;
}
30% {
  box-shadow: 9999px 0 0 2px #9880ff;
}
60%,100% {
  box-shadow: 9999px 0 0 -5px #9880ff;
}
`

const dotPulseAfter = keyframes`
0% {
  box-shadow: 10014px 0 0 -5px #9880ff;
}
30% {
  box-shadow: 10014px 0 0 2px #9880ff;
}
60%,100% {
  box-shadow: 10014px 0 0 -5px #9880ff;
}
`

export const dotPulse = css`
  position: relative;
  left: -9999px;
  text-align: initial;
  width: 10px;
  height: 10px;
  border-radius: 5px;
  background-color: #9880ff;
  color: #9880ff;
  box-shadow: 9999px 0 0 -5px #9880ff;
  animation: ${dotPulseCenter} 1.5s infinite linear;
  animation-delay: 0.25s;
  &:before,
  &:after {
    content: '';
    display: inline-block;
    position: absolute;
    top: 0;
    width: 10px;
    height: 10px;
    border-radius: 5px;
    background-color: #9880ff;
    color: #9880ff;
  }
  &:before {
    box-shadow: 9984px 0 0 -5px #9880ff;
    animation: ${dotPulseBefore} 1.5s infinite linear;
    animation-delay: 0s;
  }
  &:after {
    box-shadow: 10014px 0 0 -5px #9880ff;
    animation: ${dotPulseAfter} 1.5s infinite linear;
    animation-delay: 0.5s;
  }
`
