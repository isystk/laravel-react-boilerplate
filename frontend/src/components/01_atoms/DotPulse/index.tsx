import React, { FC } from 'react'
import { ContainerProps } from 'types'
import { connect } from '@/components/hoc'
import * as styles from './styles'

/** DotPulseProps Props */
export type DotPulseProps = Record<never, never>
/** Presenter Props */
export type PresenterProps = DotPulseProps

/** Presenter Component */
const DotPulsePresenter: FC<PresenterProps> = ({ ...props }) => (
  <>
    <div className={styles.dotPulse}></div>
  </>
)

/** Container Component */
const DotPulseContainer: React.FC<
  ContainerProps<DotPulseProps, PresenterProps>
> = ({ presenter, ...props }) => {
  return presenter({
    ...props,
  })
}

export default connect<DotPulseProps, PresenterProps>(
  'DotPulse',
  DotPulsePresenter,
  DotPulseContainer
)
