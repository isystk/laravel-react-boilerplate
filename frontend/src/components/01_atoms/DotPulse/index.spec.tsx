import React from 'react'
import renderer from 'react-test-renderer'
import DotPulse from './index'

describe('DotPulse', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<DotPulse />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
