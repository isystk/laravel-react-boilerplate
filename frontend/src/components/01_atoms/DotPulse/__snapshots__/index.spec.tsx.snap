// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DotPulse Match Snapshot 1`] = `
<div
  className="css-1oxygs2"
/>
`;
